src/main/java/hudson/plugins/git/extensions/impl/CleanBeforeCheckout.java

```diff
@@ -17,8 +17,19 @@
  * @author David S Wang
  */
 public class CleanBeforeCheckout extends GitSCMExtension {
+
+    private boolean cleanSubmodule;
     @DataBoundConstructor
+    public CleanBeforeCheckout(final boolean cleanSubmodule) {
```

Comment:
Please have a look at https://jenkins.io/doc/developer/plugin-development/pipeline-integration/#constructor-vs-setters. I suggest to not add another constructor (this procedure is hard to maintain when more and more parameters are added) but to use a setter method annotated with `@DataBoundSetter`.

Reply:
@roguishmountain did you have some time to look at this?